### PR TITLE
Akoumparouli/nemo ux ckpt conv bugfix

### PR DESCRIPTION
**Pull request by akoumpa**
# What does this PR do ?

Ensure checkpoint importer saves the model in the same precision as original checkpoint

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/lightning/pytorch/utils.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+def extract_dtypes(ckpt):
+    """
+    Extracts dtype from the input iterator
+    ckpt can be module.named_parameters or module.state_dict().items()
+    """
+    dtypes = {}
+    for key, val in ckpt:
+        try:
+            dtypes[key] = val.dtype
+        except:
```
[the remark this answers is not in the document]
Comment (cuichenx) [reply nemo/lightning/pytorch/utils.py:13]: is this to handle when val is not a tensor? @akoumpa

[the remark this answers is not in the document]
Comment (akoumpa) [reply nemo/lightning/pytorch/utils.py]: yeah, updated the code to make it obvious, I want to handle tensors and ShardedTensors.

**Review by cuichenx — APPROVED** [review]
LGTM thanks for fixing!
